doc/index.rst

```diff
@@ -0,0 +1,600 @@
+
+Welcome to typing_extensions's documentation!
+=============================================
+
+``typing_extensions`` provides common building blocks for type annotations
+that are understood by static type checkers.
+
+The ``typing_extensions`` module serves two related purposes:
+
+- Enable use of new type system features on older Python versions. For example,
+  :py:data:`typing.TypeGuard` is new in Python 3.10, but ``typing_extensions`` allows
+  users on previous Python versions to use it too.
+- Enable experimentation with new type system PEPs before they are accepted and
+  added to the :py:mod:`typing` module.
```

Comment:
"Type system PEPs" sounds slightly strange to me -- an alternative solution would be "type-system PEPs", but that also feels slightly awkward

```suggestion
- Enable experimentation with new PEPs relating to the type system before they
  are accepted and added to the :py:mod:`typing` module.
```

Reply:
How about:

```suggestion
- Enable experimentation with type system features proposed in new PEPs before they are accepted and
  added to the :py:mod:`typing` module.
```

Reply:
Yes, that works well!